fbw-common/src/systems/navdata/client/backends/Msfs/Mapping.ts

```diff
@@ -671,46 +671,54 @@ export class MsfsMapping {
-          const authorisationRequired =
-            approach.rnpAr !== undefined
-              ? approach.rnpAr
-              : approach.approachType === MSApproachType.Rnav && approach.rnavTypeFlags === 0;
-          const rnp = authorisationRequired ? 0.3 : undefined;
-          const missedApproachAuthorisationRequired =
-            approach.rnpArMissed !== undefined ? approach.rnpArMissed : authorisationRequired;
-
-          const runwayIdent = `${airportIdent}${approach.runwayNumber.toString().padStart(2, '0')}${this.mapRunwayDesignator(approach.runwayDesignator)}`;
-
-          const levelOfService = this.mapRnavTypeFlags(approach.rnavTypeFlags);
-
-          const transitions = this.mapApproachTransitions(approach, facilities, msAirport, approachName);
-
-          return {
-            sectionCode: SectionCode.Airport,
-            subSectionCode: AirportSubsectionCode.ApproachProcedures,
-            databaseId: `P${icaoCode}${airportIdent}${approach.name}`,
-            icaoCode,
-            ident: approachName,
-            suffix,
-            runwayIdent,
-            multipleIndicator: approach.approachSuffix,
-            type: this.mapApproachType(approach.approachType),
-            authorisationRequired,
-            missedApproachAuthorisationRequired,
-            levelOfService,
-            transitions,
-            legs: approach.finalLegs.map((leg, legIndex) =>
-              this.mapLeg(leg, legIndex, facilities, msAirport, approachName, approach.approachType, rnp),
-            ),
-            missedLegs: approach.missedLegs.map((leg, legIndex) =>
-              this.mapLeg(leg, legIndex, facilities, msAirport, approachName),
-            ),
-          };
+          try {
+            const approachName = this.mapApproachName(approach);
+            const suffix = approach.approachSuffix.length > 0 ? approach.approachSuffix : undefined;
+
+            // The AR flag is only available from MSFS2024, so fall back to a heuristic based on analysing the MSFS2020 data if not available.
+            const authorisationRequired =
+              approach.rnpAr !== undefined
+                ? approach.rnpAr
+                : approach.approachType === MSApproachType.Rnav && approach.rnavTypeFlags === 0;
+            const rnp = authorisationRequired ? 0.3 : undefined;
+            const missedApproachAuthorisationRequired =
+              approach.rnpArMissed !== undefined ? approach.rnpArMissed : authorisationRequired;
+
+            const runwayIdent = `${airportIdent}${approach.runwayNumber.toString().padStart(2, '0')}${this.mapRunwayDesignator(approach.runwayDesignator)}`;
+
+            const levelOfService = this.mapRnavTypeFlags(approach.rnavTypeFlags);
+
+            const transitions = this.mapApproachTransitions(approach, facilities, msAirport, approachName);
+
+            const ret: Approach = {
```

Comment:
Note: I had to give these temporary variables with explicit types, otherwise something went wrong with the filter operation later. I don't want to cast the return value because then we lose type checking for these objects.